katello/hooks/pre_commit/09-version_locking.rb

```diff
@@ -0,0 +1,21 @@
+# evaluate version locking settings
+cli_param = app_value(:lock_package_versions)
+custom_config_value = get_custom_config(:lock_package_versions)
+lock_versions = cli_param.nil? ? !!custom_config_value : cli_param
+
+if lock_versions != custom_config_value
+  store_custom_config(:lock_package_versions, lock_versions)
+  kafo.config.configure_application
+end
+
+# unlock packages if locked
+`command -v foreman-maintain`
+if $?.success?
```

Comment:
If this failed and the user indicated they want to lock this should be fatal.

Reply:
Added some more checks